module1-introduction-to-backend/r2.1-diving-into-rest-apis/README.md

```diff
@@ -0,0 +1,122 @@
+- An absolute path, ultimately followed by a '?' and query string. This is the most common form, known as the origin form, and is used with GET, POST, HEAD, and OPTIONS methods.
+```
+POST / HTTP/1.1 GET /background.png HTTP/1.0 HEAD /test.html?query=alibaba HTTP/1.1 OPTIONS /anypage.html HTTP/1.0
+```
+- A complete URL, known as the absolute form, is mostly used with GET when connected to a proxy. 
+```
+GET https://developer.mozilla.org/en-US/docs/Web/HTTP/Messages HTTP/1.1
+```
+- The authority component of a URL, consisting of the domain name and optionally the port (prefixed by a ':'), is called the authority form. It is only used with CONNECT when setting up an HTTP tunnel.
+```
+CONNECT developer.mozilla.org:80 HTTP/1.1
+```
+- The asterisk form, a simple asterisk ('*') is used with OPTIONS, representing the server as a whole.
+```
+OPTIONS * HTTP/1.1
+```
+
+### Headers
+HTTP headers from a request follow the same basic structure of an HTTP header: a case-insensitive string followed by a colon (':') and a value whose structure depends upon the header. The whole header, including the value, consist of one single line, which can be quite long.
+
+Many different headers can appear in requests. They can be divided in several groups:
+- General headers, like Via, apply to the message as a whole.
+- Request headers, like User-Agent or Accept, modify the request by specifying it further (like Accept-Language), by giving context (like Referer), or by conditionally restricting it (like If-None).
+- Representation headers like Content-Type that describe the original format of the message data and any encoding applied (only present if the message has a body).<br/><br/><img src="https://drive.google.com/uc?export=view&id=1kM2zFsDEUS6ZfHWsfMqN8LCvjM_23Fs0"><br/><br/>
+
+### Body
+The final part of the request is its body. Not all requests have one: requests fetching resources, like GET, HEAD, DELETE, or OPTIONS, usually don't need one. Some requests send data to the server in order to update it: as often the case with POST requests (containing HTML form data).
+
+Bodies can be broadly divided into two categories:
+- Single-resource bodies, consisting of one single file, defined by the two headers: Content-Type and Content-Length.
+- Multiple-resource bodies, consisting of a multipart body, each containing a different bit of information. This is typically associated with HTML Forms.
+
+## Structure of a HTTP Response
+
+The structure is similar to HTTP Requests.
+
+### Status line
+The start line of an HTTP response, called the status line, contains the following information:
+1. The protocol version, usually HTTP/1.1.
+2. A status code, indicating success or failure of the request. Common status codes are 200, 404, or 302
+3. A status text. A brief, purely informational, textual description of the status code to help a human understand the HTTP message.
+
+### Headers
```

Comment:
hm, i would consider simply making a "pointer" to the previous section rather than copy-pasting the text (it doesn't age well if you make changes to the section above), if everything is exactly the same

Reply:
It is very similar, but not exactly the same.